### ISSUE TITLE
Can't make a pure client

### ISSUE THREAD

**Issue by NKame** [quarkiverse/quarkus-cxf#281]
Context: building a command line batch in Quarkus, using quarkus-cxf for the SOAP client part.

Expectation: the batch is not a server, so it should not listen to any socket, and even less run a HTTP server.

Problem: the HTTP server is launched, setting quarkus.http.host-enabled=false fails with java.lang.IllegalArgumentException: Must configure at least one of http, https or unix domain socket

Root cause: quarkus-cxf has a mandatory dependency towards quarkus-vertx-http

**Comment by dufoli** [comment]
hmmm if we need to support both, we need to split server and client module . And of course common module.
@shumonsharif @famod @Dufgui

**Comment by famod** [comment]
A split would certainly be the clearest but also the most elaborate approach.
This might be an alternative, but I'm not sure it would mean less work because the groundwork is still a TODO: https://github.com/quarkusio/quarkus/issues/14927#issuecomment-903834399

**Comment by shumonsharif** [comment]
@NKame 's issue of disabling the HTTP server looks like it may possibly be related to a Quarkus regression:  https://github.com/quarkusio/quarkus/issues/19370

**Comment by NKame** [comment]
Well, it seems to come and go, the basis is that as soon as there is some mandatory dependency towars quarkus-vertx-http, by whatever module, the bootstrapping of HTTP server will kick in. And depending on the version, asking to disable HTTP either is almost correct (still the sequence is 90% done) or fails badly.

**Comment by NKame** [comment]
As I explained it's quite simple: depending on the versions, it's working or not, I suppose most of the users don't try to use it because of this reliability problem and put quarkus.http.port=0 to not be bothered with it. As for excluding the dependency I've done several tests: most of the times it breaks during bootstrapping.

**Comment by shumonsharif** [comment]
Thanks @NKame, I was hoping there may be some less complex workaround to address the issue.  I did some digging, and it seems the `quarkus.http.host-enabled` doesn't play nice with `quarkus-vertx-http` in the classpath.  Short of splitting server and client modules as per @dufoli , or possibly the alternative mentioned in https://github.com/quarkusio/quarkus/issues/14927#issuecomment-903834399 by @famod, I'm not sure there's much else we can do.

**Comment by NKame** [comment]
Well, it's a very low priority issue. I submitted it just in case some other people were wondering wether this has to be expected or not.

**Comment by famod** [comment]
I created https://github.com/quarkusio/quarkus/pull/20119 to fix `quarkus.http.host-enabled` w.r.t. `Must configure at least one of http, https or unix domain socket`. Let's see how it goes.

**Comment by famod** [comment]
https://github.com/quarkusio/quarkus/pull/20119 was merged and will be part of Quarkus 2.3.0 and possibly also 2.2.4.

**Comment by shumonsharif** [comment]
Quarkus-CXF 0.10.0 is out, and uses Quarkus 2.3.0.Final.

@NKame Based on @famod's fix, setting `quarkus.http.host-enabled=false` should address this issue, so I'm closing it for now, but please feel free to reopen it if you'd like to.

**Comment by famod** [comment]
I think we should add a note to the docs.

**Comment by shumonsharif** [comment]
Good point @famod , reopening this issue, will close once the docs have been updated.  Thanks for the fix!
